Week3/homework/step2-6.js

```diff
@@ -4,13 +4,27 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const flatArray2d = [];
+
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr[i].length; j++) {
+      flatArray2d.push(arr[i][j]);
+    }
+  }
+  return flatArray2d;
 }
 
 function flattenArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const flatArray3d = [];
+
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr[i].length; j++) {
+      for (let k = 0; k < arr[i][j].length; k++) {
+        flatArray3d.push(arr[i][j][k]);
+      }
+    }
+  }
+  return flatArray3d;
 }
 
 console.log(flattenArray2d(arr2d)); // -> [1, 2, 3, 4, 5, 6]
```

Comment:
These solutions are fit-for-purpose and meet the test requirements. Any idea how to deal with an array of any dimension?